src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp

```diff
@@ -42,144 +34,119 @@ enum Spells
     SPELL_SHADOWBOLT            = 17393,
     SPELL_CLEAVE                = 15284,
     SPELL_MORTALSTRIKE          = 15708,
-
-    SPELL_UNHOLY_AURA           = 15284,
-    SPELL_RAISEDEAD             = 15708,  //triggers death pact (17471)
-
+    SPELL_UNHOLY_AURA           = 17467,    // Unholy Aura is handled via DB
+    SPELL_RAISEDEAD             = 17473,    // Triggers death pact (17471)
     SPELL_RAISE_DEAD1           = 17475,
     SPELL_RAISE_DEAD2           = 17476,
     SPELL_RAISE_DEAD3           = 17477,
     SPELL_RAISE_DEAD4           = 17478,
     SPELL_RAISE_DEAD5           = 17479,
-    SPELL_RAISE_DEAD6           = 17480,
+    SPELL_RAISE_DEAD6           = 17480
 };
 
-// Define Add positions
-Position const ADD_POS_1 = {4017.403809f, -3339.703369f, 115.057655f, 5.487860f};
-Position const ADD_POS_2 = {4013.189209f, -3351.808350f, 115.052254f, 0.134280f};
-Position const ADD_POS_3 = {4017.738037f, -3363.478016f, 115.057274f, 0.723313f};
-Position const ADD_POS_4 = {4048.877197f, -3363.223633f, 115.054253f, 3.627735f};
-Position const ADD_POS_5 = {4051.777588f, -3350.893311f, 115.055351f, 3.066176f};
-Position const ADD_POS_6 = {4048.375977f, -3339.966309f, 115.055222f, 2.457497f};
+enum BaronRivendareEvents
+{
+    EVENT_SHADOWBOLT            = 1,
+    EVENT_CLEAVE                = 2,
+    EVENT_MORTALSTRIKE          = 3,
+    EVENT_RAISEDEAD             = 4
+};
 
-class boss_baron_rivendare : public CreatureScript
+struct boss_baron_rivendare : public BossAI
 {
-public:
-    boss_baron_rivendare() : CreatureScript("boss_baron_rivendare") { }
+    boss_baron_rivendare(Creature* creature) : BossAI(creature, TYPE_BARON), _raiseDead(false) { }
 
-    CreatureAI* GetAI(Creature* creature) const override
+    void Reset() override
```

Comment:
Add `BossAI::Reset();` please and you can drop `summons.DespawnAll();` and `events.Reset();`

Reply:
Won't use `BossAI::Reset();` because its important that this part won't be changed (because of https://github.com/TrinityCore/TrinityCore/issues/6377)

```
if (instance->GetData(TYPE_RAMSTEIN) == DONE)
            instance->SetData(TYPE_BARON, NOT_STARTED);
```

Reply:
adding that line won't change the SetData part